src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_ITEMS_NUMBER = 10;
+
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_ITEMS_NUMBER];
+    }
+
     @Override
     public void add(T value) {
-
+        checkToGrow();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (indexCheck(index)) {
+            checkToGrow();
+            T[] newElementData = (T[]) new Object[elementData.length];
+            System.arraycopy(elementData, 0, newElementData, 0, index);
+            newElementData[index] = value;
+            System.arraycopy(elementData, index, newElementData, index + 1, size - index);
```

Comment:
use system.arraycopy only once in each method, you can pass same array as both src and dest

Reply:
How? I've copied first part of original array then added a element and copied second part of original array to array with new element

Reply:
> use system.arraycopy only once in each method, you can pass same array as both src and dest

How? I've copied first part of original array then added a element and copied second part of original array to array with new element

Reply:
`System.arraycopy(elementData, index, elementData, index + 1, size - index);`
we don't need temp array, we copy in the same one